networkit/cpp/planarity/LeftRightPlanarityCheck.cpp

```diff
@@ -0,0 +1,262 @@
+/*  LeftRightPlanarityCheck.cpp
+ *
+ *	Created on: 03.01.2025
+ *  Authors: Andreas Scharf (andreas.b.scharf@gmail.com)
+ *
+ */
+
+#include <networkit/planarity/LeftRightPlanarityCheck.hpp>
+
+namespace NetworKit {
+
+void LeftRightPlanarityCheck::run() {
+    // Euler-criterion
+    if (graph_->numberOfNodes() > 2 && graph_->numberOfEdges() > 3 * graph_->numberOfNodes() - 6) {
+        hasRun = true;
+        isPlanar_ = false;
+        return;
+    }
+
+    heights.assign(graph_->upperNodeIdBound(), noneHeight);
+    graph_->forNodes([&](const node currentNode) {
+        if (heights[currentNode] == noneHeight) {
+            heights[currentNode] = 0;
+            roots.push_back(currentNode);
+            this->dfsOrientation(currentNode);
+        }
+    });
+
+    sortAdjacencyListByNestingDepth();
+    isPlanar_ =
+        std::ranges::all_of(roots, [this](const auto rootNode) { return dfsTesting(rootNode); });
+    hasRun = true;
+}
+
+void LeftRightPlanarityCheck::sortAdjacencyListByNestingDepth() {
+
+    dfsGraph.forNodes([&](const node currentNode) {
+        dfsGraph.sortNeighbors(currentNode, [&](const node neighbor1, const node neighbor2) {
+            if (auto it1 = nestingDepth.find(Edge(currentNode, neighbor1)),
+                it2 = nestingDepth.find(Edge(currentNode, neighbor2));
+                it1 != nestingDepth.end() && it2 != nestingDepth.end()) {
+                return it1->second < it2->second;
+            }
+            return false;
+        });
+    });
+}
+
+bool LeftRightPlanarityCheck::conflicting(const Interval &interval, const Edge &edge) {
+    return !interval.isEmpty() && lowestPoint.contains(interval.high) && lowestPoint.contains(edge)
+           && lowestPoint[interval.high] > lowestPoint[edge];
```

Comment:
Use `lowestPoint.find` instead of using `.contains` and `[]`, so that you need to do only one lookup instead of two.